ui/uihandler.go

```diff
@@ -963,28 +976,22 @@ func RedirectToHTTPSHandler(w http.ResponseWriter, r *http.Request) {
 // RedirectToHTTPS Sets up a redirector to https on the given port.
 // (Do not create a loop, make sure this is addressed from an ingress)
 func RedirectToHTTPS(port string) {
-	m := http.NewServeMux()
+	m, _ := fhttp.HTTPServer("https redirector", port)
 	m.HandleFunc("/", RedirectToHTTPSHandler)
-	s := &http.Server{
-		Addr:    fnet.NormalizePort(port),
-		Handler: m,
-	}
-	fmt.Printf("Https redirector running on %v\n", s.Addr)
-	if err := s.ListenAndServe(); err != nil {
-		log.Critf("Error starting redirect server: %v", err)
-	}
-	fmt.Printf("Not reached, https redirector exiting - was on %v\n", s.Addr)
 }
 
 // setHostAndPort takes hostport in the form of hostname:port, ip:port or :port,
-// sets the urlHostPort variable and returns hostport unmodified.
+// sets the urlHostPort variable.
 func setHostAndPort(inputPort string, addr *net.TCPAddr) {
 	urlHostPort = inputPort
 	portStr := inputPort
 	if addr != nil {
 		urlHostPort = addr.String()
 		portStr = fmt.Sprintf(":%d", addr.Port)
 	}
+	if !strings.Contains(inputPort, ":") {
```

Comment:
call NormalizePort?

Reply:
yeah the logic is a bit in 2 places but I didn't want to re-call that method; I think I need to cleanup the whole uihandler some more in the future - right now just trying to get manual tests to work with localhost special case

Reply:
ok